django/db/views.py

```diff
@@ -2328,27 +2333,29 @@ def export_as_xls(name,col_key_name_map, cursor=None, queryset=None,
     elif queryset:
         for obj in queryset:  
             if isinstance(obj, dict): # a ORM object as a dict
-                vals = [_write_val_safe(obj[field]) \
+                vals = [_write_val_safe(obj[field]) 
                             for field in col_key_name_map.keys()]
             else: # a ORM object
                 vals = [getattr(obj,field) for field in col_key_name_map.keys()]
             
-            for i,column in enumerate(vals):
+            temp = []
+            for column in vals:
                 # if the columnn is a method, we are referencing the method 
                 # wrapper for restricted columns
                 if(inspect.ismethod(column)):
-                    sheet.write(
-                        row+1,i, _write_val_safe(
-                            column(is_authenticated=is_authenticated)) )
+                    temp.append(_write_val_safe(
+                        column(is_authenticated=is_authenticated)) )
                 else:
-                    sheet.write(row+1, i, _write_val_safe(column) )
+                    temp.append(_write_val_safe(column))
+            ws.append(temp)
             if(row % debug_interval == 0):
                 logger.info("row: " + str(row))
             row += 1    
-    
-    wbk.save(response)
-    return response
 
+    response = HttpResponse(save_virtual_workbook(wb), content_type='application/vnd.ms-excel')
+    response['Content-Disposition'] = 'attachment; filename=%s.xlsx' % unicode(name).replace('.', '_')
```

Comment:
yes - it is totally under our control - and should always be composed of ascii chars (based on the code in place), so this is meant as a catch all/final check. Ideally, a constraint (assertion?) would be imposed - ?